lib/grandpa/message_handler.go

```diff
@@ -80,28 +80,26 @@ func (h *MessageHandler) handleMessage(from peer.ID, m GrandpaMessage) (network.
 }
 
 func (h *MessageHandler) handleNeighbourMessage(msg *NeighbourPacketV1) error {
-	if h.grandpa.authority {
-		// TODO(#2931): this is a simple hack to ensure that the neighbour messages
-		// sent by gossamer are being received by substrate nodes
-		// not intended to be production code
-		h.grandpa.roundLock.Lock()
-		neighbourMessage := &NeighbourPacketV1{
-			Round:  h.grandpa.state.round,
-			SetID:  h.grandpa.state.setID,
-			Number: uint32(h.grandpa.head.Number),
-		}
-		h.grandpa.roundLock.Unlock()
-
-		cm, err := neighbourMessage.ToConsensusMessage()
-		if err != nil {
-			return fmt.Errorf("converting neighbour message to network message: %w", err)
-		}
-
-		logger.Debugf("sending neighbour message: %v", neighbourMessage)
-		h.grandpa.network.GossipMessage(cm)
+	// TODO(#2931): this is a simple hack to ensure that the neighbour messages
+	// sent by gossamer are being received by substrate nodes
+	// not intended to be production code
+	round := h.blockState.GetRound()
+	setID := h.blockState.GetSetID()
```

Comment:
should we check that these aren't zero values?

Reply:
we don't need to lock the mutex anymore?

Reply:
hmm I am not sure that we need to add checks. We were actually only using zero values before, and from my tests I know that the values do progress so I don't think we need to error out or anything.

As to locks, the locking still happens I moved it to internal locking instead of having the caller handle it, which I thought was cleaner.

Reply:
I think you should return both round and setID in one call then, and do internal locking around that.  You have the risk of the round or set ID changing between these two functions calls.

Reply:
good call, updated